src/wasm-lib/kcl/src/settings/types/mod.rs

```diff
@@ -278,8 +278,65 @@ pub struct ModelingSettings {
     /// Whether or not to show a scale grid in the 3D modeling view
     #[serde(default, alias = "showScaleGrid", skip_serializing_if = "is_default")]
     pub show_scale_grid: bool,
+    /// Settings that affect the behavior of the command bar.
+    #[serde(default, alias = "namedViews")]
+    #[validate(nested)]
+    pub named_views: Vec<NamedView>,
 }
 
+
+fn one () -> f64 {
+    return 1.0
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize, JsonSchema, ts_rs::TS, Validate)]
+#[serde(rename_all = "snake_case")]
+#[ts(export)]
+pub struct NamedView {
+    /// The default unit to use in modeling dimensions.
+    #[serde(default, alias = "name", skip_serializing_if = "is_default")]
+    pub name: String,
+    /// The default unit to use in modeling dimensions.
+    #[serde(default, alias = "eyeOffset", skip_serializing_if = "is_default")]
+    pub eye_offset: f64,
+    /// The default unit to use in modeling dimensions.
+    #[serde(default, alias = "fovY", skip_serializing_if = "is_default")]
+    pub fov_y: f64,
+    /// The default unit to use in modeling dimensions.
+    #[serde(default, alias = "isOrtho")]
+    pub is_ortho: bool,
+    /// The default unit to use in modeling dimensions.
+    #[serde(default, alias = "orthoScaleEnabled")]
+    pub ortho_scale_enabled: bool,
+    /// The default unit to use in modeling dimensions.
+    #[serde(default, alias = "orthoScaleFactor", skip_serializing_if = "is_default")]
+    pub ortho_scale_factor: f64,
+    /// The default unit to use in modeling dimensions.
+    #[serde(default, alias = "pivotPosition", skip_serializing_if = "is_default")]
+    pub pivot_position: [f64; 3],
+    /// The default unit to use in modeling dimensions.
+    #[serde(default, alias = "pivotRotation", skip_serializing_if = "is_default")]
+    pub pivot_rotation: [f64; 4],
+    /// The default unit to use in modeling dimensions.
+    #[serde(default, alias = "worldCoordSystem", skip_serializing_if = "is_default")]
+    pub world_coord_system: String,
+    /// The default unit to use in modeling dimensions.
+    #[serde(default = "uuid::Uuid::new_v4", alias = "_id")]
+    pub _id: uuid::Uuid,
```

Comment:
id not prefix w an underscore in rust, thats weird, the alias will make sure it parses

Reply:
Not just that but it's semantically incorrect in Rust right? In Rust, `_var` means it's an unused variable. I'm pretty sure it's "nonsensical" to say this for a struct property? Could someone correct me?

Reply:
You are correct, Lee. In Rust, `_name` tells the compiler that it's an unused variable/field/parameter and no warning should be emitted if nothing uses it.